jmsSource/src/main/java/io/vantiq/extsrc/jmsSource/communication/JMSMessageProducer.java

```diff
@@ -0,0 +1,120 @@
+     * @param connectionFactoryName     The name of the connection factory used to connect to the JMS Server
+     * @param dest                      The name of the destination to connect to (topic or queue)
+     * @param isQueue                   A boolean flag used to create the appropriate type of destination (queue or topic)
+     * @param username                  The username used to create the JMS Connection, (or null if JMS Server does not require auth)
+     * @param password                  The password used to create the JMS Connection, (or null if JMS Server does not require auth)
+     * @throws NamingException
+     * @throws JMSException
+     */
+    public void setupMessageProducer(String connectionFactoryName, String dest, boolean isQueue, String username, String password) throws NamingException, JMSException {
+        this.destName = dest;
+        connectionFactory = (ConnectionFactory) context.lookup(connectionFactoryName);
+        connection = connectionFactory.createConnection(username, password);
+        session = connection.createSession(false, Session.AUTO_ACKNOWLEDGE);
+        if (isQueue) {
+            destination = session.createQueue(dest);
+        } else {
+            destination = session.createTopic(dest);
+        }
+        producer = session.createProducer(destination);
+        producer.setDeliveryMode(DeliveryMode.NON_PERSISTENT);
+        connection.start();
+    }
+    
+    /**
+     * Called by the JMS Class, and used to send the provided message to the associated destination, (topic or queue),
+     * using whatever format was specified
+     * @param message           The message to be sent
+     * @param messageFormat     The format (JMS Message Type) of the message to be sent
+     * @throws JMSException
+     */
+    public void produceMessage(String message, String messageFormat) throws JMSException {
+        switch(messageFormat) {
+            case MESSAGE:
+                Message baseMessage = session.createMessage();
+                producer.send(baseMessage);
+                break;
+            case BYTES:
+                // Handle it
+                BytesMessage byteMessage = session.createBytesMessage();
+                break;
+            case TEXT:
+                TextMessage textMessage = session.createTextMessage();
+                textMessage.setText(message);
+                producer.send(textMessage);
+                break;
+            case STREAM:
+                // Handle it
+                StreamMessage streamMessage = session.createStreamMessage();
+                break;
+            case MAP:
+                // Handle it
+                MapMessage mapMessage = session.createMapMessage();
+                break;
+            case OBJECT:
+                // Handle it
+                ObjectMessage objectMessage = session.createObjectMessage();
+                break;
+        }
```

Comment:
need `default:` case for blown software.  Or JMS may evolve or some provider might implement `fredsWonderFormat` which we don't support and should note...